lib/Service/RoomService.php

```diff
@@ -392,4 +409,101 @@ public function verifyPassword(Room $room, string $password): array {
 			'url' => '',
 		];
 	}
+
+	public function setTimeToLive(Room $room, Participant $participant, int $ttl): void {
+		$event = new ChangeTtlEvent($room, $ttl);
+		$this->dispatcher->dispatch(Room::EVENT_BEFORE_SET_TIME_TO_LIVE, $event);
+
+		$update = $this->db->getQueryBuilder();
+		$update->update('talk_rooms')
+			->set('time_to_live', $update->createNamedParameter($ttl, IQueryBuilder::PARAM_INT))
+			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
+		$update->executeStatement();
+		$jobList = Server::get(IJobList::class);
+		if ($ttl > 0) {
+			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_enabled');
+			$jobList->add(ApplyTtl::class, ['room_id' => $room->getId()]);
+		} else {
+			$this->ttlSystemMessage($room, $participant, $ttl, 'ttl_disabled');
+			$jobList->remove(ApplyTtl::class, ['room_id' => $room->getId()]);
+		}
+
+		$this->dispatcher->dispatch(Room::EVENT_AFTER_SET_TIME_TO_LIVE, $event);
+	}
+
+	private function ttlSystemMessage(Room $room, Participant $participant, int $ttl, string $message): void {
+		$this->chatManager->addSystemMessage(
+			$room,
+			$participant->getAttendee()->getActorType(),
+			$participant->getAttendee()->getActorId(),
+			json_encode([
+				'message' => $message,
+				'parameters' => ['ttl' => $ttl]
+			]),
+			$this->timeFactory->getDateTime(),
+			false
+		);
+	}
+
+
+	public function deleteExpiredTtl(int $roomId, int $jobId): array {
+		$room = $this->manager->getRoomById($roomId);
+
+		$max = $this->getMaxDateTtl($room->getTimeToLive());
+		$min = $this->getMinDateTtl($jobId);
+
+		$ids = $this->commentsManager->getMessageIdsByRoomIdInDateInterval($roomId, $min, $max);
+		if (count($ids)) {
```

Comment:
If you are not interested in the value, don't count:
```suggestion
		if (!empty($ids)) {
```